src/test/java/tests/MultiModelTest.java

```diff
@@ -25,4 +39,21 @@ public void multiModelTest() {
         assertEquals(new MultiModel(new Preferences(true)).hashCode(), multiModel.hashCode());
     }
 
+    @Test
+    public void testRemoveModel() throws ExecutionException, InterruptedException {
+        final String repoId = "dummy1/dummy1";
+        MultiModel models = new MultiModel(new Preferences(true));
+        models.queuePendingRepository(repoId);
+        RepoIO testIO = new RepoIO(true, false);
+        testIO.openRepository(repoId).thenApply(models::addPending);
+
+        TestUtils.delay(3);
```

Comment:
`sleep` in tests is bad, as it may cause randomness and timing issues. Is it possible to replace this with `PlatformEx.waitOnFxThread()`?